### PR TITLE
server _routeErrorResponse  avoiding uncaughtException by only sending response when not sent

### PR DESCRIPTION
**Pull request by beckend**
middlewares and routes which send response normally, however server listeners such as MethodNotAllowed will trigger the route error handler and send response regardless if payload has been sent or not and causes an uncaughtException needlessly.

Closes:

* Issue #1567

### CODE REVIEW

**Comment by hekike** [comment]
Thanks for the contribution!

I applied a very similar fix on the 7.x branch: https://github.com/restify/node-restify/pull/1561 with https://github.com/restify/node-restify/pull/1561/commits/e31564172a82b886613961e40da9d3fb88d30256 and I think it's also related to the https://github.com/restify/node-restify/issues/1557

@beckend could you please add a test to cover this case?

**Comment by beckend** [comment]
@hekike test proven

File: test/server.test.js [hunk 1]
```diff
@@ -224,6 +224,33 @@ test('rm route and clear cached route', function(t) {
     });
 });
 
+test('_routeErrorResponse does not cause uncaughtException when called when header has already been sent', function(
+    t
+) {
+    SERVER.on('MethodNotAllowed', function(req, res, error, next) {
+        res.json(405, { status: 'MethodNotAllowed' });
+        var nextHasErrored;
```
Comment (hekike) [inline test/server.test.js]: You can do this:

```js
try {
    next();
} catch (err) {
    t.fail('next() should not throw error when header has already been sent');
}
t.end();
```

**Comment by hekike** [comment]
@beckend thanks, a small comment, otherwise looks great!

**Comment by hekike** [comment]
@beckend thanks for your contribution! Landed as `v6.3.4`.

**Pull request merged**